vault/activity_log_util_common.go

```diff
@@ -288,3 +290,95 @@ func (a *ActivityLog) mountAccessorToMountPath(mountAccessor string) string {
 	}
 	return displayPath
 }
+
+type singleTypeSegmentReader struct {
```

Comment:
One last question, are these structs and helper functions are going to be used only in tests or in the production code as well?

Reply:
in production code, the next PR will refactor the precomputed query worker to call this.

Reply:
Great! The reason I asked for it was that if they are going to be used only in testing, maybe it would be good to move them to another place! Thanks for the context!